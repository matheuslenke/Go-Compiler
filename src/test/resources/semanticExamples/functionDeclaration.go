package sample

import "fmt"

func func1() {
	var s string = "func1"
	fmt.Println(s)
	return
}

func func2() string {
	func1()
	var s string = "func1"
	fmt.Println(s)

	return s
}

func func3(a, b string) int {
	// var resultF1 string = func2()
	var s3, s4 string = "func1", "Hello world" + "t"

	var returnVar int = 5
	return returnVar
}

func func4() {
	var s5 = 1
}