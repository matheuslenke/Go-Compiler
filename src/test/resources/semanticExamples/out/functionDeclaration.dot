digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) func1@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="var_list"];
node14[label="="];
node15[label="(string) s@6"];
node14 -> node15;
node16[label="(string) string@0"];
node14 -> node16;
node13 -> node14;
node12 -> node13;
node17[label="write"];
node18[label="Println@1"];
node17 -> node18;
node19[label="arguments"];
node20[label="(string) s@6"];
node19 -> node20;
node17 -> node19;
node12 -> node17;
node21[label="return"];
node12 -> node21;
node10 -> node12;
node0 -> node10;
node22[label="(func) func2@7"];
node23[label="parameters"];
node22 -> node23;
node24[label="result"];
node25[label="(string) type_use"];
node24 -> node25;
node22 -> node24;
node26[label="block"];
node27[label="func_use"];
node28[label="(func) func1@5"];
node27 -> node28;
node29[label="arguments"];
node27 -> node29;
node26 -> node27;
node30[label="var_list"];
node31[label="="];
node32[label="(string) s@8"];
node31 -> node32;
node33[label="(string) string@0"];
node31 -> node33;
node30 -> node31;
node26 -> node30;
node34[label="write"];
node35[label="Println@1"];
node34 -> node35;
node36[label="arguments"];
node37[label="(string) s@8"];
node36 -> node37;
node34 -> node36;
node26 -> node34;
node38[label="return"];
node39[label="(string) s@8"];
node38 -> node39;
node26 -> node38;
node22 -> node26;
node0 -> node22;
node40[label="(func) func3@9"];
node41[label="parameters"];
node42[label="(string) a@10"];
node41 -> node42;
node43[label="(string) b@11"];
node41 -> node43;
node40 -> node41;
node44[label="result"];
node45[label="(int) type_use"];
node44 -> node45;
node40 -> node44;
node46[label="block"];
node47[label="var_list"];
node48[label="="];
node49[label="(string) s3@12"];
node48 -> node49;
node50[label="(string) string@0"];
node48 -> node50;
node47 -> node48;
node51[label="="];
node52[label="(string) s4@13"];
node51 -> node52;
node53[label="(string) +"];
node54[label="(string) string@1"];
node53 -> node54;
node55[label="(string) string@2"];
node53 -> node55;
node51 -> node53;
node47 -> node51;
node46 -> node47;
node56[label="var_list"];
node57[label="="];
node58[label="(int) returnVar@14"];
node57 -> node58;
node59[label="(int) int@5"];
node57 -> node59;
node56 -> node57;
node46 -> node56;
node60[label="return"];
node61[label="(int) returnVar@14"];
node60 -> node61;
node46 -> node60;
node40 -> node46;
node0 -> node40;
node62[label="(func) func4@15"];
node63[label="parameters"];
node62 -> node63;
node64[label="block"];
node65[label="var_list"];
node66[label="="];
node67[label="(int) s5@16"];
node66 -> node67;
node68[label="(int) int@1"];
node66 -> node68;
node65 -> node66;
node64 -> node65;
node62 -> node64;
node0 -> node62;
}
