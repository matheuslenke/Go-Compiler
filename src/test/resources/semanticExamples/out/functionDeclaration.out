[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: func1
Entry 1: Hello world
Entry 2: t




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: func1, line: 5, type: func, scopeName: global
Entry 6 -- name: s, line: 6, type: string, scopeName: func1
Entry 7 -- name: func2, line: 11, type: func, scopeName: global
Entry 8 -- name: s, line: 13, type: string, scopeName: func2
Entry 9 -- name: func3, line: 19, type: func, scopeName: global
Entry 10 -- name: a, line: 19, type: string, scopeName: func3
Entry 11 -- name: b, line: 19, type: string, scopeName: func3
Entry 12 -- name: s3, line: 21, type: string, scopeName: func3
Entry 13 -- name: s4, line: 21, type: string, scopeName: func3
Entry 14 -- name: returnVar, line: 23, type: int, scopeName: func3
Entry 15 -- name: func4, line: 27, type: func, scopeName: global
Entry 16 -- name: s5, line: 28, type: int, scopeName: func4




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.850 s
[INFO] Finished at: 2022-02-24T20:53:48-03:00
[INFO] ------------------------------------------------------------------------
